tests/cpp-tests/Classes/SpriteFrameCacheTest/SpriteFrameCacheTest.cpp

```diff
@@ -0,0 +1,85 @@
+
+#include "SpriteFrameCacheTest.h"
+
+// enable log
+#define COCOS2D_DEBUG 1
+
+USING_NS_CC;
+
+SpriteFrameCacheTests::SpriteFrameCacheTests()
+{
+    ADD_TEST_CASE(SpriteFrameCachePixelFormatTest);
+}
+
+SpriteFrameCachePixelFormatTest::SpriteFrameCachePixelFormatTest()
+{
+    const Size screenSize = Director::getInstance()->getWinSize();
+
+    infoLabel = Label::create();
+    infoLabel->setAnchorPoint(Point(0.5f, 1.0f));
+    infoLabel->setAlignment(cocos2d::TextHAlignment::CENTER);
+    infoLabel->setPosition(screenSize.width * 0.5f, screenSize.height * 0.7f);
+    addChild(infoLabel);
+    
+    // load atlas definition with specified PixelFormat and check that it matches to expected format
+    loadSpriteFrames("Images/sprite_frames_test/test_A8.plist", Texture2D::PixelFormat::A8);
+    loadSpriteFrames("Images/sprite_frames_test/test_RGBA8888.plist", Texture2D::PixelFormat::RGBA8888);
+    loadSpriteFrames("Images/sprite_frames_test/test_AI88.plist", Texture2D::PixelFormat::AI88);
+    loadSpriteFrames("Images/sprite_frames_test/test_RGBA8888.plist", Texture2D::PixelFormat::RGBA8888);
+    loadSpriteFrames("Images/sprite_frames_test/test_RGB565.plist", Texture2D::PixelFormat::RGB565);
+    loadSpriteFrames("Images/sprite_frames_test/test_RGB888.plist", Texture2D::PixelFormat::RGB888);
+    loadSpriteFrames("Images/sprite_frames_test/test_RGBA4444.plist", Texture2D::PixelFormat::RGBA4444);
+    loadSpriteFrames("Images/sprite_frames_test/test_RGBA5551.plist", Texture2D::PixelFormat::RGB5A1);
+#if CC_TARGET_PLATFORM == CC_PLATFORM_IOS || CC_TARGET_PLATFORM == CC_PLATFORM_MAC
+    loadSpriteFrames("Images/sprite_frames_test/test_PVRTC2.plist", Texture2D::PixelFormat::PVRTC2A);
+    loadSpriteFrames("Images/sprite_frames_test/test_PVRTC4.plist", Texture2D::PixelFormat::PVRTC4A);
+    loadSpriteFrames("Images/sprite_frames_test/test_PVRTC2_NOALPHA.plist", Texture2D::PixelFormat::PVRTC2);
+#endif
+    
+    // test loading atlases wihtout PixelFormat specified
+    Texture2D::setDefaultAlphaPixelFormat(Texture2D::PixelFormat::RGB5A1);
+    loadSpriteFrames("Images/sprite_frames_test/test_NoFormat.plist", Texture2D::PixelFormat::RGB5A1);
+    
+    // restore default alpha pixel format
+    Texture2D::setDefaultAlphaPixelFormat(Texture2D::PixelFormat::RGBA8888);
+}
+
+void SpriteFrameCachePixelFormatTest::loadSpriteFrames(const std::string &file, cocos2d::Texture2D::PixelFormat expectedFormat)
+{
+    SpriteFrameCache::getInstance()->addSpriteFramesWithFile(file);
+    SpriteFrame *spriteFrame = SpriteFrameCache::getInstance()->getSpriteFrameByName("grossini.png");
+    Texture2D *texture = spriteFrame->getTexture();
+    const ssize_t bitsPerKB = 8 * 1024;
+    const double memorySize = 1.0 * texture->getBitsPerPixelForFormat() * texture->getContentSizeInPixels().width * texture->getContentSizeInPixels().height / bitsPerKB;
+    CC_ASSERT(texture->getPixelFormat() == expectedFormat);
```

Comment:
`texture->getPixelFormat() != expectedFormat` when `expectedFormat` is `Texture2D::PixelFormat::PVRTCXX`